docs/Configurable.md

```diff
@@ -0,0 +1,123 @@
+---
+layout: default
+title: "The Configurable Component"
+---
+
+# `<Configurable>`
+
+This component makes another component configurable by the end user. When they enter the configuration mode, users can customize the component's settings via the inspector.
+
+![SimpleListConfigurable](./img/SimpleListConfigurable.gif)
+
+Some react-admin components are already configurable - or rather they have a configurable counterpart:
+
+- [`<SimpleListConfigurable>`](./SimpleList.md#configurable)
+
+## Usage
+
+Wrap any component with `<Configurable>` and define its editor to let users customize it via a UI. Don't forget to pass down props to the inner component. Note that every configurable component needs a unique preference key, that is used to persist the user's preferences in the Store.
+
+```jsx
+import { Configurable } from 'react-admin';
+
+const ConfigurableTextBlock = ({ preferenceKey = "textBlock", ...props }) => (
+    <Configurable editor={<TextBlockEditor />} preferenceKey={preferenceKey}>
+        <TextBlock {...props} />
+    </Configurable>
+);
+```
+
+`<Configurable>` creates a context for the `preferenceKey`, so that both the child component and the editor can access it using `usePreferenceKey()`.
+ 
+Then, use this component in your app:
+
+```jsx
+import { ConfigurableTextBlock } from './ConfigurableTextBlock';
```

Comment:
> since there are a lot of custom components scattered around here (they are all defined, yes, but not in readable order, which may be confusing).

Not sure I understand your point. Could you please rephrase?

Reply:
For example, the `TextBlockInspector` component reference is not defined anywhere